cloud/scope/machine.go

```diff
@@ -275,19 +275,20 @@ func (m *MachineScope) GetVMID() string {
 
 // GetProviderID returns the AzureMachine providerID from the spec.
 func (m *MachineScope) GetProviderID() string {
-	if m.AzureMachine.Spec.ProviderID != nil {
-		return *m.AzureMachine.Spec.ProviderID
+	parsed, err := noderefutil.NewProviderID(to.String(m.AzureMachine.Spec.ProviderID))
+	if err != nil {
+		return ""
 	}
-	return ""
+	return parsed.ID()
 }
 
 // SetProviderID sets the AzureMachine providerID in spec.
 func (m *MachineScope) SetProviderID(v string) {
 	m.AzureMachine.Spec.ProviderID = to.StringPtr(v)
 }
 
-// GetVMState returns the AzureMachine VM state.
-func (m *MachineScope) GetVMState() infrav1.VMState {
+// VMState returns the AzureMachine VM state.
+func (m *MachineScope) VMState() infrav1.VMState {
```

Comment:
Assuming you're changing this to align with getter style for Go, do you want to change `GetProviderID()` to `ProviderID()` too?

Reply:
renamed